models/classes/notification/implementation/AbstractSqlNotificationService.php

```diff
@@ -0,0 +1,227 @@
+            $userId     = $notificationDetail[self::NOTIFICATION_FIELD_RECIPIENT];
+            $title      = $notificationDetail[self::NOTIFICATION_FIELD_TITLE];
+            $message    = $notificationDetail[self::NOTIFICATION_FIELD_MESSAGE];
+            $senderId   = $notificationDetail[self::NOTIFICATION_FIELD_SENDER];
+            $senderName = $notificationDetail[self::NOTIFICATION_FIELD_SENDER_NAME];
+            $id         = $notificationDetail[self::NOTIFICATION_FIELD_ID];
+            $createdAt  = $notificationDetail[self::NOTIFICATION_FIELD_CREATION];
+            $updatedAt  = $notificationDetail[self::NOTIFICATION_FIELD_UPDATED];
+            $status     = $notificationDetail[self::NOTIFICATION_FIELD_STATUS];
+            $notification[] = new Notification($userId, $title, $message, $senderId, $senderName, $id, $createdAt, $updatedAt, $status);
+        }
+
+        return $notification;
+    }
+
+    protected function getPersistence(): Persistence
+    {
+        if ($this->persistence === null) {
+            $persistence = self::DEFAULT_PERSISTENCE;
+
+            if ($this->hasOption(self::OPTION_PERSISTENCE)) {
+                $persistence = $this->getOption(self::OPTION_PERSISTENCE);
+            }
+
+            $persistenceManager = $this->getServiceLocator()->get(PersistenceManager::SERVICE_ID);
+            $this->persistence  = $persistenceManager->getPersistenceById($persistence);
+        }
+        return $this->persistence;
+    }
+
+    /**
+     * @throws common_exception_NotFound
+     */
+    public function getNotification(string $id): Notification
+    {
+        /** @var common_persistence_SqlPersistence $persistence */
+        $persistence = $this->getPersistence();
+
+        $selectQuery = 'SELECT ' . self::NOTIFICATION_FIELD_ID . ' , ' . $this->getAllFieldString() .
+            ' FROM ' . self::NOTIFICATION_TABLE . ' WHERE ' .
+            self::NOTIFICATION_FIELD_ID . ' = ? ';
+
+        $params = [
+            $id,
+        ];
+
+        $notificationDetail = $persistence->query($selectQuery, $params)->fetch();
+
+        if ($notificationDetail) {
+            $userId = $notificationDetail[self::NOTIFICATION_FIELD_RECIPIENT];
+            $title = $notificationDetail[self::NOTIFICATION_FIELD_TITLE];
+            $message = $notificationDetail[self::NOTIFICATION_FIELD_MESSAGE];
+            $senderId = $notificationDetail[self::NOTIFICATION_FIELD_SENDER];
+            $id = $notificationDetail[self::NOTIFICATION_FIELD_ID];
+            $createdAt = $notificationDetail[self::NOTIFICATION_FIELD_CREATION];
+            $updatedAt = $notificationDetail[self::NOTIFICATION_FIELD_UPDATED];
+            $status = $notificationDetail[self::NOTIFICATION_FIELD_STATUS];
+
+            return new Notification($userId, $title, $message, $senderId, $id, $createdAt, $updatedAt, $status);
+        }
+
+        throw new common_exception_NotFound('unknown notification id ' . $id);
```

Comment:
as a improvement i would change the message  the message

Reply:
```suggestion
throw new common_exception_NotFound('Notification not found');
```

Reply:
that's perfect , but i think we should leave the Id, 
```suggestion
        throw new common_exception_NotFound('Error notification not found ,  requested id: ' . $id);
```